taskcluster/docker/node/build.py

```diff
@@ -63,6 +64,54 @@ def get_package_info():
     return contents
 
 
+def write_package_info(package_info):
+    with open("package.json", "w") as fh:
+        json.dump(package_info, fh)
+    assert get_package_info() == package_info
+
+
+def get_buildid():
+    now = datetime.utcnow()
+    return now.strftime("%Y%m%d%H%M%S")
+
+
+def get_buildid_version(version):
+    """Version schema check+append a `buildid{buildid}` to ensure unique version."""
+    # XXX is there a more precise schema we should verify ourselves against?
+    if len(version.split(".")) not in (1, 2, 3):
+        raise Exception("{version} has too many version parts!")
+    if "buildid" in version:
+        raise Exception("{version} already has a buildid specified!")
+    buildid_version = f"{version}buildid{get_buildid()}"
```

Comment:
Some questions in this function:

- Is there a more precise schema we can use to verify our version? (And should I verify it before or after appending `buildid{buildid}`?)
- Right now we're going with up to 3 version parts, and appending `buildid20211011135501` or similar... looks like all our existing privileged/system/mozillaonline-privileged addons fit in the 3-or-fewer version parts bucket at this point. Should I keep the long buildid, or add a `.` between the date and time, as we did in [this patch](https://phabricator.services.mozilla.com/D126815)?

Reply:
To be safe you need the same logic from that patch otherwise you risk running into the same issues we did for langpacks where version comparison would fail if somehow only the `buildid` changes. It's probably a good idea to be completely consistent with what we are going to do for langpacks moving forward anyway.